mergify_engine/tests/functional/base.py

```diff
@@ -614,12 +618,36 @@ async def create_pr(
         else:
             open(self.git.tmp + f"/test{self.pr_counter}", "wb").close()
             await self.git("add", f"test{self.pr_counter}")
-        await self.git("commit", "--no-edit", "-m", title)
+        args_commit = [
+            "commit",
+            "--no-edit",
+            "-m",
+            title,
+        ]
+        if verified:
+            temporary_folder = tempfile.mkdtemp()
+            self.addCleanup(shutil.rmtree, temporary_folder)
+            tmp_env = os.environ.copy()
+            tmp_env["GNUPGHOME"] = temporary_folder
+            subprocess.run(
+                ["gpg", "--import"], input=config.TESTING_GPGKEY_SECRET, env=tmp_env
+            )
+            await self.git("config", "user.signingkey", config.TESTING_ID_GPGKEY_SECRET)
+            await self.git(
+                "config", "user.email", "engineering+mergify-test@mergify.io"
+            )
+            args_commit.append("-S")
```

Comment:
I'm surprised that you don't need to pass the env to the git command.
You should remove the key from your local gpg keyring and retry to record it, it should not pass.